### PR TITLE
Add bottommost_compression_opts to for bottommost_compression

### PR DESCRIPTION
**Pull request by zhichao-cao**
…ression

Summary: For `CompressionType` we have options `compression` and `bottommost_compression`. Thus, to make the compression options consitent with the compression type when bottommost_compression is enabled, we add the bottommost_compression_opts

Test Plan: Added the configuration test

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: options/options_helper.cc [hunk 1]
```diff
@@ -904,6 +904,63 @@ Status ParseColumnFamilyOption(const std::string& name,
...
+      // enabled is optional for backwards compatibility
+      if (end != std::string::npos) {
+        start = end + 1;
+        if (start >= value.size()) {
+          return Status::InvalidArgument(
+              "unable to parse the specified CF option " + name);
+        }
+        new_options->bottommost_compression_opts.enabled =
+            ParseBoolean("", value.substr(start, value.size() - start));
+      }
```
Comment (ajkr) [inline options/options_helper.cc]: can you extract this to a function since it's exactly the same logic as `compression_opts`?

File: include/rocksdb/options.h [hunk 2]
```diff
@@ -221,6 +221,10 @@ struct ColumnFamilyOptions : public AdvancedColumnFamilyOptions {
   // Default: kDisableCompressionOption (Disabled)
   CompressionType bottommost_compression = kDisableCompressionOption;
 
+  // different options for compression algorithms used by bottommost_compression
+  // if it is enabled
```
Comment (ajkr) [inline include/rocksdb/options.h]: can we be more explicit about what does "enabled" mean here? Maybe point the user to the `CompressionOpts::enabled` comment and thoroughly document it there.

File: include/rocksdb/advanced_options.h [hunk 3]
```diff
@@ -126,19 +126,28 @@ struct CompressionOptions {
   // Default: 0.
   uint32_t zstd_max_train_bytes;
 
+  // When the compression options are set by the user, it will be set to "true".
```
Comment (ajkr) [inline include/rocksdb/advanced_options.h]: let's make it explicit that the user must set `enabled=true` in order for `bottommost_compression_opts` to be used. And even if `compression_opts.enabled=false`, we will still use it.

File: db/db_test2.cc [hunk 4]
```diff
@@ -1084,6 +1088,9 @@ TEST_F(DBTest2, PresetCompressionDict) {
           options.compression_opts.max_dict_bytes = kBlockSizeBytes;
           options.compression_opts.zstd_max_train_bytes =
               kZstdTrainFactor * kBlockSizeBytes;
+          options.bottommost_compression_opts.max_dict_bytes = kBlockSizeBytes;
```
Comment (ajkr) [inline db/db_test2.cc]: are these changes needed? It looks like we're not setting `enabled=true` here so the values in `bottommost_compression_opts` shouldn't be used, right?

File: options/options_helper.cc (hunk 1 above)
Comment (zhichao-cao) [reply options/options_helper.cc]: Sure, it will shorter the length a lot.

File: include/rocksdb/options.h (hunk 2 above)
Comment (zhichao-cao) [reply include/rocksdb/options.h]: I think it will be better to point out that if they enable it and what to specify the options for bottommost compression, they should look at the bottommost_compression_opts and CompressionOptions definition.

File: db/db_test2.cc (hunk 4 above)
Comment (zhichao-cao) [reply db/db_test2.cc]: Yeah.....I will remove them

File: HISTORY.md [hunk 5]
```diff
@@ -3,6 +3,7 @@
 ### Public API Change
 * For users of `Statistics` objects created via `CreateDBStatistics()`, the format of the string returned by its `ToString()` method has changed.
 * With LRUCache, when high_pri_pool_ratio > 0, midpoint insertion strategy will be enabled to put low-pri items to the tail of low-pri list (the midpoint) when they first inserted into the cache. This is to make cache entries never get hit age out faster, improving cache efficiency when large background scan presents.
+* For bottommost_compression, a compatable compression options is added via `bottommost_compression_opts`. To keep backward compatible, a new boolean `enabled` is added to CompressionOptions. For compression_opts, it will be always used no matter what value of `enabled` is. For bottommost_compression_opts, it will only be used when user set `enabled=true`, otherwise, compression_opts will be used for bottommost_compression as default.
```
Comment (ajkr) [inline HISTORY.md]: s/compatable/compatible/

Comment (zhichao-cao) [reply HISTORY.md]: Oh, sorry about the typo, I will correct it. Thanks for the review.
